src/Model/Model.ts

```diff
@@ -622,7 +665,18 @@ export class BaseModel {
   }
 
   public toJSON(): unknown {
-    return this.$attributes;
+    const Model = this.constructor as typeof BaseModel;
+
+    const computed: Record<string, unknown> = {};
+    for (const key of Model.$computedDefinitions.keys()) {
+      // @ts-expect-error polymorphic getter
+      computed[key] = this[key];
+    }
```

Comment:
You need to ignore keys if `serializeAs` is `null`

Reply:
Done, full support of serializeAs.

* `null` ignore serialization
* non-empty string use it as key
* falsy value fallback on getter name